x-pack/plugins/synthetics/server/legacy_uptime/lib/saved_objects/migrations/monitors/8.8.0.ts

```diff
@@ -118,3 +126,73 @@ const omitZipUrlFields = (fields: BrowserFields) => {
 
   return formatSecrets(validationResult.decodedMonitor);
 };
+
+const updateThrottlingFields = (
+  doc: SavedObjectUnsanitizedDoc<
+    SyntheticsMonitorWithSecrets &
+      Partial<{
+        [LegacyConfigKey.THROTTLING_CONFIG]: string;
+        [LegacyConfigKey.IS_THROTTLING_ENABLED]: boolean;
+        [LegacyConfigKey.DOWNLOAD_SPEED]: string;
+        [LegacyConfigKey.UPLOAD_SPEED]: string;
+        [LegacyConfigKey.LATENCY]: string;
+      }>
+  >
+) => {
+  const { attributes } = doc;
+
+  const migrated = {
+    ...doc,
+    attributes: {
+      ...doc.attributes,
+      [ConfigKey.CONFIG_HASH]: '',
+    },
+  };
+
+  const isThrottlingEnabled = attributes[LegacyConfigKey.IS_THROTTLING_ENABLED];
+  if (isThrottlingEnabled) {
+    const download = String(attributes[LegacyConfigKey.DOWNLOAD_SPEED])!;
+    const upload = String(attributes[LegacyConfigKey.UPLOAD_SPEED])!;
+    const latency = String(attributes[LegacyConfigKey.LATENCY])!;
+
+    const isDefault = isDefaultThrottlingConfig(download, upload, latency);
+
+    const newThrottlingConfig: ThrottlingConfig = {
+      value: {
+        download,
+        upload,
+        latency,
+      },
+      id: isDefault ? PROFILE_VALUES_ENUM.DEFAULT : 'custom',
```

Comment:
Should we use some type of constant for the `custom` id to prevent typos? Also for the labels below, potentially.